smart_card_connector_app/src/application_unittest.cc

```diff
@@ -79,19 +75,82 @@ class SmartCardConnectorApplicationTest : public ::testing::Test {
+  // when a new reader starts to be initialized).
+  const std::vector<std::string>& reader_init_add_messages() const {
+    return reader_init_add_messages_;
+  }
+  // Returns all "reader_finish_add" messages that were sent to JS (they are
+  // sent when a reader initialization finishes).
+  const std::vector<std::string>& reader_finish_add_messages() const {
+    return reader_finish_add_messages_;
+  }
+
+ private:
+  void SetUpUsbSimulation() {
+    global_context_.RegisterRequestHandler(
+        TestingSmartCardSimulation::kRequesterName,
+        std::bind(&TestingSmartCardSimulation::OnRequestToJs,
+                  &smart_card_simulation_,
+                  /*request_payload=*/std::placeholders::_1,
+                  /*request_id=*/std::placeholders::_2));
+  }
+
+  void SetUpReaderMessageObserving() {
+    global_context_.RegisterMessageHandler(
+        "reader_init_add",
+        std::bind(&SmartCardConnectorApplicationTest::OnReaderInitAddPosted,
+                  this, /*request_payload=*/std::placeholders::_1,
+                  /*request_id=*/std::placeholders::_2));
+    global_context_.RegisterMessageHandler(
+        "reader_finish_add",
+        std::bind(&SmartCardConnectorApplicationTest::OnReaderFinishAddPosted,
+                  this, /*request_payload=*/std::placeholders::_1,
+                  /*request_id=*/std::placeholders::_2));
+  }
+
+  // Called when the code-under-test sends a "reader_init_add" message to JS.
+  void OnReaderInitAddPosted(Value request_payload,
+                             optional<RequestId> /*request_id*/) {
+    reader_init_add_messages_.push_back(
+        request_payload.GetDictionaryItem("readerName")->GetString());
+  }
+
+  // Called when the code-under-test sends a "reader_finish_add" message to JS.
+  void OnReaderFinishAddPosted(Value request_payload,
+                               optional<RequestId> /*request_id*/) {
+    reader_finish_add_messages_.push_back(
+        request_payload.GetDictionaryItem("readerName")->GetString());
+  }
+
   TypedMessageRouter typed_message_router_;
   TestingSmartCardSimulation smart_card_simulation_{&typed_message_router_};
   TestingGlobalContext global_context_{&typed_message_router_};
   std::unique_ptr<Application> application_;
+  std::vector<std::string> reader_init_add_messages_;
+  std::vector<std::string> reader_finish_add_messages_;
 };
 
-TEST_F(SmartCardConnectorApplicationTest, SmokeTest) {}
+TEST_F(SmartCardConnectorApplicationTest, SmokeTest) {
+  StartApplication();
+}
 
 // Test a PC/SC-Lite context can be established and freed, via direct C function
 // calls `SCardEstablishContext()` and `SCardReleaseContext()`.
 // This is an extended version of the "smoke test" as it verifies the daemon
 // successfully started and replies to calls sent over (fake) sockets.
 TEST_F(SmartCardConnectorApplicationTest, InternalApiContextEstablishing) {
+  // Arrange:
+  StartApplication();
```

Comment:
nit: missing an empty line between comment and function call for consistency (or remove the empty lines on the other Act/Arranges, I don't think they are necessary)